### ISSUE TITLE
Problem with Dataloader and labels

### ISSUE THREAD

**Issue by Krutsch** [pytorch/pytorch#22566]
Hello, I am classifying images and realized something disturbing... 
While the test data achieved 90% accuracy with the help of the dataloader API, single predictions were often wrong, so I iterated through my images without the dataloader API and saw that the trained model only achieved 15% accuracy.
So I was wondering why..
I am using the oxford_102 dataset which I load locally by using `torchvision.datasets.ImageFolder` and `torch.utils.data.DataLoader`. In my train folder are 102 subfolder - from 0 to 101. 

The Problem in the train method:

```
for images, labels in dataloaders[phase]: 
    print(labels)
    imshow(torchvision.utils.make_grid(images))  # method [here](https://pytorch.org/tutorials/beginner/blitz/cifar10_tutorial.html)
```

Images from folder **0** and **1** are displayed correctly. When the label **2** gets printed, I can see images from folder **10**. **So I am guessing that the loader is taking the correct order for the labels, but the wrong order for the images...**
 

System:
 - PyTorch 1.1 with pip
 - Windows 10
 - python 3.7.3
 - CUDA v10.1.168
 - GTX 1080

**Comment by Krutsch** [comment]
I found a solution for it by changing my evalution method:
```
out = model(batch.to(device))
_, pred = torch.max(out, 1)

len_classes = len(np.ndarray.flatten(out.cpu().detach().numpy()))
index_to_cat = [str(x) for x in range(len_classes)]
index_to_cat.sort()
index_to_cat = [int(x) for x in index_to_cat]

prediction = index_to_cat[pred]
correct_label = int(os.path.basename(os.path.dirname(path)))
```

But I still think the API methods should do it for me

**Comment by SsnL** [comment]
This is about torchvision's ImageFolder. Could you open the issue there instead? 

cc @fmassa

**Comment by fmassa** [comment]
@Krutsch I believe this is the same issue as https://github.com/pytorch/vision/issues/714

I'm closing this issue as it's not a PyTorch problem. If you want to continue the discussion, open an issue in torchvision.
